paris-tennis.py

```diff
@@ -0,0 +1,178 @@
+BOOKING_PAGE = 'https://tennis.paris.fr/tennis/jsp/site/Portal.jsp?page=recherche&view=recherche_creneau#!'
+PLAYER_USERNAME = "myEmail@gmail.com"
+PLAYER_PASSWORD = "myPassword"
+PARTNER_NAME = 'myPartner'
+PARTNER_EMAIL = 'mypartner@gmail.com'
+
+HOLD_MODE = 'hold'
+PAY_MODE = 'pay'
+BOT_MODES = [PAY_MODE, HOLD_MODE]
+BotMode = HOLD_MODE  # by default only hold the court
+
+
+def login(username, password):
+    try:
+        logging.info("Logging in")
+        options = webdriver.ChromeOptions()
+        driver = webdriver.Chrome(ChromeDriverManager().install())
+        driver.get(LOGIN_PAGE)
+        window_before = driver.window_handles[0]
+        driver.find_element_by_id('button_suivi_inscription').click()
+        time.sleep(0.5)
+        # LOGIN PAGE
+        window_after = driver.window_handles[1]
+        driver.switch_to.window(window_after)
+        time.sleep(0.5)
+        driver.find_element_by_id('username-login').send_keys(username)
+        driver.find_element_by_id('password-login').send_keys(password)
+        time.sleep(0.5)
+        driver.find_element_by_name('Submit').click()
+        time.sleep(1)
+        driver.switch_to.window(window_before)
+        time.sleep(0.4)
+        driver.get(BOOKING_PAGE)
+        time.sleep(0.5)
+        return driver
+    except Exception as e:
+        logging.error("Failure to login", e)
+
+
+def pick_tennis_court(driver, day, name):
+    try:
+        time.sleep(0.5)
+        driver.find_element_by_id('whereToken').click()
+        driver.find_element_by_class_name('tokens-input-text').send_keys(name)
+        time.sleep(1)
+        driver.find_element_by_class_name('tokens-suggestions-list-element').click()
+
+        # FIND SLOT
+        driver.find_element_by_id('when').click()
+        time.sleep(0.5)
+        driver.find_elements_by_class_name('date-item')[day].click()
+        driver.find_element_by_id('rechercher').click()
+        time.sleep(1)
+    except Exception as e:
+        logging.error("Failure to pick a tennis court", e)
+
+
+def book_and_pay(driver, hour, day, name):
+    try:
+        # Click on the arrow to make the booking rows available
+        driver.find_element_by_id('head{}{}'.format(name, hour)).click()
+        time.sleep(2)
+
+        # Click Reserver button
+        driver.find_element_by_id('collapse{}{}'.format(name, hour)).find_elements_by_class_name('tennis-court')[
+            0].find_element_by_class_name('button').click()
+        time.sleep(1)
+        # FILL PARTNER INFOS
+        driver.find_element_by_class_name('name').find_element_by_class_name(
+            'form-control').send_keys(PARTNER_NAME)
+        time.sleep(1)
+        driver.find_element_by_class_name('firstname').find_element_by_class_name(
+            'form-control').send_keys(PARTNER_EMAIL)
+        time.sleep(1)
+        driver.find_element_by_id('submitControle').click()
+        time.sleep(1)
+
+        if BotMode is not HOLD_MODE:
```

Comment:
It might be better to move the .notify() invocation outside the if/else statement since all the parameters (except message) are the same for both modes. I guess setting the message in the if/else statement and in some variable and using that in the .notify() invocation afterwards.

Reply:
I can add a function with default params

`def sys_notify(message, title='Paris Sport Bot', timeout=10)`

WDYT?